src/Servers/HttpSys/src/NativeInterop/HttpApi.cs

```diff
@@ -55,18 +55,22 @@ internal unsafe delegate uint HttpGetRequestPropertyInvoker(SafeHandle requestQu
 
     static unsafe HttpApi()
     {
-        var statusCode = PInvoke.HttpInitialize(Version, HTTP_INITIALIZE.HTTP_INITIALIZE_SERVER | HTTP_INITIALIZE.HTTP_INITIALIZE_CONFIG);
-
-        if (statusCode == ErrorCodes.ERROR_SUCCESS)
+        try
         {
-            Supported = true;
-            HttpApiModule = SafeLibraryHandle.Open(HTTPAPI);
-            HttpGetRequestProperty = HttpApiModule.GetProcAddress<HttpGetRequestPropertyInvoker>("HttpQueryRequestProperty", throwIfNotFound: false);
-            HttpSetRequestProperty = HttpApiModule.GetProcAddress<HttpSetRequestPropertyInvoker>("HttpSetRequestProperty", throwIfNotFound: false);
-            SupportsReset = HttpSetRequestProperty != null;
-            SupportsTrailers = IsFeatureSupported(HTTP_FEATURE_ID.HttpFeatureResponseTrailers);
-            SupportsDelegation = IsFeatureSupported(HTTP_FEATURE_ID.HttpFeatureDelegateEx);
+            var statusCode = PInvoke.HttpInitialize(Version, HTTP_INITIALIZE.HTTP_INITIALIZE_SERVER | HTTP_INITIALIZE.HTTP_INITIALIZE_CONFIG);
+
+            if (statusCode == ErrorCodes.ERROR_SUCCESS)
+            {
+                Supported = true;
+                HttpApiModule = SafeLibraryHandle.Open(HTTPAPI);
+                HttpGetRequestProperty = HttpApiModule.GetProcAddress<HttpGetRequestPropertyInvoker>("HttpQueryRequestProperty", throwIfNotFound: false);
+                HttpSetRequestProperty = HttpApiModule.GetProcAddress<HttpSetRequestPropertyInvoker>("HttpSetRequestProperty", throwIfNotFound: false);
+                SupportsReset = HttpSetRequestProperty != null;
+                SupportsTrailers = IsFeatureSupported(HTTP_FEATURE_ID.HttpFeatureResponseTrailers);
+                SupportsDelegation = IsFeatureSupported(HTTP_FEATURE_ID.HttpFeatureDelegateEx);
+            }
         }
+        catch (DllNotFoundException) { }
```

Comment:
`HttpApi.SupportsDelegation` is used to skip tests via https://github.com/dotnet/aspnetcore/blob/3babc099d656a297b45e16e52abce4a2fa0810ff/src/Servers/HttpSys/test/NonHelixTests/DelegateSupportedConditionAttribute.cs#L14

This of course throws on non-Windows platforms and it looks like xunit now reports the exception during test discovery instead of silently skipping it.

Reply:
Sounds good, done!